turbo/snapshotsync/freezeblocks/block_reader.go

```diff
@@ -379,7 +397,7 @@ func (r *BlockReader) HeaderByNumber(ctx context.Context, tx kv.Getter, blockHei
 	}
 
 	if tx != nil {
-		blockHash, err := rawdb.ReadCanonicalHash(tx, blockHeight)
+		blockHash, err := r.CanonicalHash(ctx, tx, blockHeight)
```

Comment:
seems here can use `rawdb.ReadCanonicalHash`

Reply:
mh, I prefer using `r.CanonicalHash`, whenever I can, I just should regardless. but yes I see your point, but this is just to be safe and consistent